elastic/kibana
# [App Search] Add new encodePathParams helper (fixes unencoded document IDs)

## Summary

I discovered this issue while playing around with analytics queries with special characters - I realized requests being sent to the Enterprise Search were 404ing due to unencoded URL parameters.

### Reproducing the issue

- Go to http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/documents (or whatever other engine)
- Create the following JSON document:
```json
[
  {
    "id": "test!@#*$(&!)@(*$@][]/hmm{}",
    "title": "test"
  }
]
```
- Refresh the page and click into the new test document
- Confirm you see the following error:
  <img width="788" alt="" src="https://user-images.githubusercontent.com/549407/105012153-b1825b00-59f2-11eb-9624-020e7733dcd0.png">
- Go to http://localhost:3002/as/engines/national-parks-demo/documents/
- Click the result for `test!@#*$(&!)@(*$@][]/hmm{}`
- Confirm you do not see an error

### Confirming the fix

- Checkout this branch
- Go to http://localhost:5601/xyz/app/enterprise_search/app_search/engines/national-parks-demo/documents
- Click the result for `test!@#*$(&!)@(*$@][]/hmm{}`
- Confirm you are now correctly taken to document detail view:
  <img width="808" alt="" src="https://user-images.githubusercontent.com/549407/105012413-f9a17d80-59f2-11eb-8a8b-3ff3f0471a26.png">

## QA

- [x] Confirm the above fix works
- Confirm all other API endpoints updated to use the new helper have not regressed:
- [x] Engines Overview renders normally
- [x] Engine Overview renders normally
- [x] Settings (get + put) works w/o errors
- [x] Credentials - creating, deleting, editing works w/o errors

### Checklist
